crates/anvil-polkadot/tests/it/utils.rs

```diff
@@ -0,0 +1,146 @@
+
+        let (anvil_config, mut substrate_config) = anvil_args.node.clone().into_node_config()?;
+        let anvil_config = anvil_config.set_silent(true);
+
+        let temp_dir = tempfile::tempdir()?;
+        let db_path = temp_dir.path().join("db");
+
+        substrate_config.set_base_path(Some(db_path));
+        let config = substrate_config.create_configuration(&anvil_args, handle.clone())?;
+        let service = service::new(&anvil_config, config)?;
+        let api = spawn_anvil_tasks(anvil_config, &service).await?;
+        tokio::task::yield_now().await;
+
+        Ok(Self { service, api, _runtime_handle: handle.clone() })
+    }
+
+    pub async fn call_eth(&mut self, req: EthRequest) -> Result<ResponseResult> {
+        let (tx, rx) = oneshot::channel();
+        self.api
+            .try_send(api_server::ApiRequest { req: req.clone(), resp_sender: tx })
+            .map_err(|e| eyre::eyre!("failed to send EthRequest {:?}: {}", req, e))?;
+
+        rx.await.map_err(|e| eyre::eyre!("ApiRequest receiver dropped: {}", e))
+    }
+
+    async fn call_rpc(&self, method: &str, params: Value) -> Result<Value> {
+        let rpc = &self.service.rpc_handlers;
+
+        let request = json!({
+            "jsonrpc": "2.0",
+            "method": method,
+            "params": params,
+            "id": 1
+        });
+
+        let (response, _receiver) = rpc
+            .rpc_query(&request.to_string())
+            .await
+            .wrap_err(format!("RPC call failed for method: {method}"))?;
+
+        let response_value: Value =
+            serde_json::from_str(&response).wrap_err("Failed to parse RPC response")?;
+
+        if let Some(error) = response_value.get("error") {
+            return Err(eyre::eyre!("RPC error: {}", error));
+        }
+
+        response_value
+            .get("result")
+            .cloned()
+            .ok_or_else(|| eyre::eyre!("No result in RPC response"))
+    }
+
+    async fn call_rpc_no_params(&self, method: &str) -> Result<Value> {
+        self.call_rpc(method, json!([])).await
+    }
+}
+
+impl TestNode {
+    pub async fn system_chain(&self) -> Result<String> {
+        let result = self.call_rpc_no_params("system_chain").await?;
+        Ok(result.as_str().unwrap_or("").to_string())
+    }
+
+    pub async fn get_best_block_number(&self) -> Result<u32> {
+        let best_number = self.service.client.info().best_number;
+        Ok(best_number)
+    }
+
+    pub async fn block_hash_by_number(&self, n: u32) -> eyre::Result<H256> {
+        self.service
+            .client
+            .block_hash(n)
+            .wrap_err("client.block_hash failed")?
+            .ok_or_else(|| eyre::eyre!("no hash for block {}", n))
+    }
+
+    pub fn create_storage_key(pallet: &str, item: &str) -> StorageKey {
+        let mut key = Vec::new();
+        key.extend_from_slice(&twox_128(pallet.as_bytes()));
+        key.extend_from_slice(&twox_128(item.as_bytes()));
+        StorageKey(key)
+    }
+
```

Comment:
not all storage keys are created in this way, but it's fine for now. We can add modify this when needed